gwtquery-core/src/main/java/com/google/gwt/query/client/plugins/effects/Transform.java

```diff
@@ -0,0 +1,165 @@
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package com.google.gwt.query.client.plugins.effects;
+
+import com.google.gwt.dom.client.Element;
+import com.google.gwt.query.client.GQuery;
+import com.google.gwt.regexp.shared.MatchResult;
+import com.google.gwt.regexp.shared.RegExp;
+
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map.Entry;
+
+/**
+ * A dictionary class with all the properties of an element transform
+ * which is able to return the correct syntax for setting the css transform
+ * property.
+ */
+public class Transform  {
+
+  private static final String TRANSFORM = "_t_";
+
+  protected static final RegExp transformRegex = RegExp.compile("^(scale([XYZ]|)|translate([XYZ]|3d)|rotate([XYZ]|3d)?|perspective|skew[XYZ]|x|y)$");
+  private static final RegExp transform3dRegex = RegExp.compile("^(rotate([XYZ]|3d)|perspective)$");
+
+  private HashMap<String, List<String>> map = new HashMap<String, List<String>>();
+
+  /**
+   * Return the Transform dictionary object of a element.
+   */
+  public static Transform getInstance(Element e) {
+    return getInstance(e, null);
+  }
+
+  public static boolean isTransform(String propName) {
+    return transformRegex.test(propName);
+  }
+
+  /**
+   * Return the Transform dictionary object of an element, but reseting
+   * historical values and setting them to the initial value passed.
+   */
+  public static Transform getInstance(Element e, String initial) {
+    Transform t = GQuery.data(e, TRANSFORM);
+    if (t == null || initial != null && !initial.isEmpty()) {
+      t = new Transform(initial);
+      GQuery.data(e, TRANSFORM, t);
+    }
+    return t;
+  }
+
+  public Transform(String s) {
+    parse(s);
+  }
+
+  public String get(String prop) {
+    return listToStr(map.get(prop), ",");
+  }
+
+  private String listToStr(List<String> l, String sep) {
+    String v = "";
+    if (l != null) {
+      for (String s : l) {
+        v += (v.isEmpty() ? "" : sep) + s;
+      }
+    }
+    return v;
+  }
+
+  private void parse(String s) {
+    if (s != null) {
+      RegExp re = RegExp.compile("([a-zA-Z0-9]+)\\((.*?)\\)", "g");
+      for (MatchResult r = re.exec(s); r != null; r = re.exec(s)) {
+        setFromString(r.getGroup(1), r.getGroup(2));
+      }
+    }
+  }
+
+  public void set(String prop, String ...val) {
+    setter(prop, val);
+  }
+
+  public void setFromString(String prop, String ...val) {
+    if (val.length == 1) {
+      String[] vals = val[0].split("[\\s*,\\s*]");
+      set(prop, vals);
+    } else {
+      set(prop, val);
+    }
+  }
+
+  private void setter(String prop, String ...val) {
+    if (prop.matches("(rotate[XYZ]?|skew[XYZ])")) {
```

Comment:
Reviewed and tested all regexp, I think this code was ported from original plugin and most regex were not tested nor optimised enough